### PR TITLE
Separate date input

### PR DESCRIPTION
**Pull request by frederike-ramin**
# Short Description
- We separate the date field into day, month and year input fields.

# Changes
- Add another MultipleInputFieldWidget class
- Add the possibility to set field_labels to MultipleInputFieldWidget itself
- Adapt tests (dates are now sent as array, not as string)

# Feedback
- What do you think of the way I added the sub_field_labels?
- Do you see how we could get rid of the adaption of `render_kw` in `SteuerlotseDateField.__init()__`?

### CODE REVIEW

File: webapp/tests/forms/flows/test_unlock_code_revocation_flow.py [hunk 1]
```diff
@@ -179,8 +179,8 @@ def test_if_user_exists_and_dob_correct_and_unlock_code_revocation_got_through_t
 
     def test_if_user_exists_and_dob_correct_and_unlock_code_revocation_got_through_then_user_is_deleted(self):
         existing_idnr = '04452397687'
-        correct_dob = '01.01.1985'
-        create_user(existing_idnr, correct_dob, '0000')
+        correct_dob = ['1', '1', '1985']
```
Comment (JulianRoesner) [inline webapp/tests/forms/flows/test_unlock_code_revocation_flow.py:182]: Why can't we keep the data as a string? Didn't we do it like that for the idnr?

File: webapp/tests/forms/test_fields.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        self.form.process()
+        self.assertEqual([], self.form.date_field._value())
+
+    def test_if_data_given_and_no_formdata_then_value_equals_list_of_data(self):
+        """ Simulates a GET request with prefilled data."""
+        prefilled_date = dt.date(1979, 9, 19)
+        self.form.process(data={'date_field': prefilled_date})
+        self.assertEqual([19, 9, 1979], self.form.date_field._value())
+
+    def test_if_no_data_given_and_invalid_formdata_then_value_equals_formdata_as_is(self):
```
Comment (JulianRoesner) [inline webapp/tests/forms/test_fields.py:75]: I think this is exactly what I meant. I think I'm still a little confused on how we are going to use these separated fields in the future.

File: webapp/app/forms/fields.py [hunk 3]
```diff
@@ -63,6 +72,37 @@ def _value(self):
...
+class SteuerlotseDateField(DateField):
+
+    def __init__(self, **kwargs):
+        kwargs.setdefault('format', "%d %m %Y")
+
+        if kwargs.get('render_kw'):
+            if kwargs['render_kw'].get('class'):
+                kwargs['render_kw']['class'] = kwargs['render_kw']['class'] + " date_input form-control"
+            else:
+                kwargs['render_kw']['class'] = "date_input form-control"
```
Comment (JulianRoesner) [inline webapp/app/forms/fields.py:90]: We could get rid of these lines:
```suggestion
           kwargs['render_kw']['class'] = kwargs['render_kw'].get('class', '') + " date_input form-control"
```

File: webapp/app/forms/fields.py [hunk 4]
```diff
@@ -63,6 +72,37 @@ def _value(self):
...
+            if 'example_input' not in kwargs['render_kw']:
+                kwargs['render_kw']['example_input'] = _('fields.date_field.example_input.text')
```
Comment (JulianRoesner) [inline webapp/app/forms/fields.py:92]: and also of this conditional
```suggestion
                kwargs['render_kw']['example_input'] = kwargs['render_kw'].get('example_input', _('fields.date_field.example_input.text'))
```

**Review by JulianRoesner — APPROVED** [review]
I think I am still not really getting the separated fields. I only see some improvements for the `render_kw` part. Unfortunately, I do not see a solution to reach the same goals but without setting the render_kw in the init

File: webapp/tests/forms/flows/test_unlock_code_revocation_flow.py (hunk 1 above)
Comment (frederike-ramin) [reply webapp/tests/forms/flows/test_unlock_code_revocation_flow.py:182]: Mh, no. This is the input as given by the form. And the form - in both cases, idnr and dob, sends the data as a list of strings. Only difference is (and that is why I did not have to change this test for idnr) that the validator of date relies on a special representation (the list), while the idnr validator can handle both string and list.

I'd be hesitant to store the date as string and here's why:
- for IdNr we have two representations: `list` if not (yet) valid or in the form, `string` if valid or in the stored_data
- for date we'd have three represenations: `list` if in the form, `string` if not (yet) valid and `date` if valid or in the stored_data
Do you disagree?

File: webapp/tests/forms/test_fields.py (hunk 2 above)
Comment (frederike-ramin) [reply webapp/tests/forms/test_fields.py:75]: What do you mean is what you meant? :P

File: webapp/app/forms/fields.py (hunk 3 above)
Comment (frederike-ramin) [reply webapp/app/forms/fields.py:90]: Niiice!

File: webapp/app/forms/fields.py (hunk 4 above)
Comment (frederike-ramin) [reply webapp/app/forms/fields.py:92]: Wow, I am impressed. 👍

File: webapp/tests/forms/flows/test_unlock_code_revocation_flow.py (hunk 1 above)
Comment (JulianRoesner) [reply webapp/tests/forms/flows/test_unlock_code_revocation_flow.py:182]: Thanks, everything makes sense now

File: webapp/tests/forms/test_fields.py (hunk 2 above)
Comment (JulianRoesner) [reply webapp/tests/forms/test_fields.py:75]: See comment above about list, string and date

File: webapp/app/forms/fields.py (hunk 4 above)
Comment (JulianRoesner) [reply webapp/app/forms/fields.py:92]: 😊

**Pull request merged**